pysph/examples/solid_mech/oscillating_plate.py

```diff
@@ -0,0 +1,170 @@
+                wdeltap=self.wdeltap, n=4, rho_ref=self.plate_rho0,
+                E=self.plate_E, nu=self.plate_nu, amplitude_idx=amplitude_idx))
+
+        ##################################
+        # vertical velocity of the plate #
+        ##################################
+        # initialize with zero at the beginning
+        v = np.zeros_like(xp)
+        v = v.ravel()
+
+        # set the vertical velocity for particles which are only
+        # out of the wall
+        K = self.K
+        # L = self.L
+        KL = self.KL
+        M = sin(KL) + sinh(KL)
+        N = cos(KL) + cosh(KL)
+        Q = 2 * (cos(KL) * sinh(KL) - sin(KL) * cosh(KL))
+        for i in range(len(v)):
+            if xp[i] > 0.:
+                # set the velocity
+                tmp1 = (cos(K * xp[i]) - cosh(K * xp[i]))
+                tmp2 = (sin(K * xp[i]) - sinh(K * xp[i]))
+                v[i] = self.Vf * plate.cs[0] * (M * tmp1 - N * tmp2) / Q
+
+        # set vertical velocity
+        plate.v = v
+
+        # #########################################
+        # #### Create the wall particle array #####
+        # #########################################
+        # get the minimum and maximum of the plate
+        xp_min = xp.min()
+        yp_min = yp.min()
+        yp_max = yp.max()
+        xw_upper, yw_upper = np.mgrid[-self.plate_inside_wall_length:self.
+                                      dx_plate / 2.:self.dx_plate, yp_max +
+                                      self.dx_plate:yp_max + self.dx_plate +
+                                      (self.wall_layers - 1) * self.dx_plate +
+                                      self.dx_plate / 2.:self.dx_plate]
+        xw_upper = xw_upper.ravel()
+        yw_upper = yw_upper.ravel()
+
+        xw_lower, yw_lower = np.mgrid[-self.plate_inside_wall_length:self.
+                                      dx_plate / 2.:self.dx_plate, yp_min -
+                                      self.dx_plate:yp_min - self.dx_plate -
+                                      (self.wall_layers - 1) * self.dx_plate -
+                                      self.dx_plate / 2.:-self.dx_plate]
+        xw_lower = xw_lower.ravel()
+        yw_lower = yw_lower.ravel()
+
+        xw_left_max = xp_min - self.dx_plate
+        xw_left_min = xw_left_max - (
+            self.wall_layers - 1) * self.dx_plate - self.dx_plate / 2.
+        yw_left_max = yw_upper.max() + self.dx_plate / 2.
+        yw_left_min = yw_lower.min()
+
+        xw_left, yw_left = np.mgrid[xw_left_max:xw_left_min:-self.dx_plate,
+                                    yw_left_min:yw_left_max:self.dx_plate]
+        xw_left = xw_left.ravel()
+        yw_left = yw_left.ravel()
+
+        # wall coordinates
+        xw, yw = np.concatenate((xw_lower, xw_upper, xw_left)), np.concatenate(
+            (yw_lower, yw_upper, yw_left))
+
+        # create the particle array
+        wall = get_particle_array_elastic_dynamics(
```

Comment:
you should use simple `get_particle_array()` function here and add desired properties using 'self.scheme.setup_properties()' this will make the scope of adding more schemes in future.